src/Avalonia.Controls/Primitives/UniformGrid.Properties.cs

```diff
@@ -0,0 +1,146 @@
+        internal static void SetAutoLayout(ColumnDefinition element, bool? value)
+        {
+            element.SetValue(AutoLayoutProperty, value);
+        }
+
+        /// <summary>
+        /// Gets the AutoLayout Attached Property Value. Used internally to track items which need to be arranged vs. fixed in place.
+        /// </summary>
+        /// <param name="element"><see cref="ColumnDefinition"/></param>
+        /// <returns>A true value indicates this item should be automatically arranged.</returns>
+        internal static bool? GetAutoLayout(ColumnDefinition element)
+        {
+            return element.GetValue(AutoLayoutProperty);
+        }
+
+        /// <summary>
+        /// Sets the AutoLayout Attached Property Value. Used internally to track items which need to be arranged vs. fixed in place.
+        /// </summary>
+        /// <param name="element"><see cref="RowDefinition"/></param>
+        /// <param name="value">A true value indicates this item should be automatically arranged.</param>
+        internal static void SetAutoLayout(RowDefinition element, bool? value)
+        {
+            element.SetValue(AutoLayoutProperty, value);
+        }
+
+        /// <summary>
+        /// Gets the AutoLayout Attached Property Value. Used internally to track items which need to be arranged vs. fixed in place.
+        /// </summary>
+        /// <param name="element"><see cref="RowDefinition"/></param>
+        /// <returns>A true value indicates this item should be automatically arranged.</returns>
+        internal static bool? GetAutoLayout(RowDefinition element)
+        {
+            return element.GetValue(AutoLayoutProperty);
+        }
+
+        /// <summary>
+        /// Identifies the <see cref="Columns"/> dependency property.
+        /// </summary>
+        public static readonly StyledProperty<int> ColumnsProperty =
+            AvaloniaProperty.Register<UniformGrid, int>(nameof(Columns), 0);
+
+        /// <summary>
+        /// Identifies the <see cref="FirstColumn"/> dependency property.
+        /// </summary>
+        public static readonly StyledProperty<int> FirstColumnProperty =
+            AvaloniaProperty.Register<UniformGrid, int>(nameof(FirstColumn), 0);
+
+        /// <summary>
+        /// Identifies the <see cref="Orientation"/> dependency property.
+        /// </summary>
+        public static readonly StyledProperty<Orientation> OrientationProperty =
+            AvaloniaProperty.Register<UniformGrid, Orientation>(nameof(Orientation), Orientation.Horizontal);
+
+        /// <summary>
+        /// Identifies the <see cref="Rows"/> dependency property.
+        /// </summary>
+        public static readonly StyledProperty<int> RowsProperty =
+            AvaloniaProperty.Register<UniformGrid, int>(nameof(Rows), 0);
+
+        /// <summary>
+        /// Gets or sets the number of columns in the UniformGrid.
+        /// </summary>
+        public int Columns
```

Comment:
I've always thought this should be named `ColumnCount`. `Columns` implies it is taking objects like Grid. I don't know why the naming was decided to be what it is in the WCT.

Reply:
This might be even more relevant now that you are inheriting from Grid.

Reply:
Well, it will be double breaking change, because our old UniformGrid had this naming, and this new UniformGrid from WCT has exact same naming.

Reply:
If its already in Avalonia, nevermind. I doubt I can successfully argue this justifies a breaking change.